core/domainstorage/storage.go

```diff
@@ -12,8 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package adminstorage stores multi-tennancy configuration information.
-package adminstorage
+// Package domainstorage stores multi-tennancy configuration information.
+package domainstorage
```

Comment:
This name leads to what the Go folk call 'stuttering' in package.ExportedType names:
`domainstorage.Storage`

Reply:
Hmm, good point. Would `domain.Storage` and `domain.Domain` be better?